src/ui/LiveButton.js

```diff
@@ -11,48 +11,78 @@ import {
 } from '@apollosproject/ui-kit';
 import { AnalyticsConsumer } from '@apollosproject/ui-analytics';
 import { PLAY_VIDEO } from '@apollosproject/ui-media-player';
-import { LiveConsumer } from '@apollosproject/ui-connected';
+import {
+  LiveConsumer,
+  RockAuthedWebBrowser,
+} from '@apollosproject/ui-connected';
 
 const LiveCard = styled(({ theme }) => ({
   backgroundColor: theme.colors.lightSecondary,
 }))(Card);
 
+const renderLiveButton = ({ action }) => (
+  <TouchableScale onPress={action}>
+    <LiveCard>
+      <CardContent>
+        <ChannelLabel
+          icon="video"
+          label={
+            <UIText>
+              <UIText bold>{`We're live.`} </UIText>
+              Watch now!
+            </UIText>
+          }
+        />
+      </CardContent>
+    </LiveCard>
+  </TouchableScale>
+);
+
+renderLiveButton.propTypes = {
+  action: PropTypes.func,
+};
+
+const renderMedia = (liveStream, track) => (
+  <Mutation mutation={PLAY_VIDEO}>
+    {(play) =>
+      renderLiveButton({
+        action: () => {
+          play({
+            variables: {
+              mediaSource: liveStream.media.sources[0],
+              isVideo: true,
+            },
+          });
+          track({ eventName: 'Clicked Live Bar' });
+        },
+      })
+    }
+  </Mutation>
+);
+
+const renderWebView = (liveStream, track) => (
+  <RockAuthedWebBrowser>
+    {(openUrl) =>
+      renderLiveButton({
+        action: () => {
+          openUrl(liveStream.webViewUrl);
+          track({ eventName: 'Clicked Live Bar' });
+        },
+      })
+    }
+  </RockAuthedWebBrowser>
+);
+
 const LiveNowButton = ({ contentId }) => (
   <LiveConsumer contentId={contentId}>
     {(liveStream) =>
       liveStream ? (
         <AnalyticsConsumer>
-          {({ track }) => (
-            <Mutation mutation={PLAY_VIDEO}>
-              {(play) => (
-                <TouchableScale
-                  onPress={() => {
-                    play({
-                      variables: {
-                        mediaSource: liveStream.media.sources[0],
-                        isVideo: true,
-                      },
-                    });
-                    track({ eventName: 'Clicked Live Bar' });
-                  }}
-                >
-                  <LiveCard>
-                    <CardContent>
-                      <ChannelLabel
-                        icon="video"
-                        label={
-                          <UIText>
-                            <UIText bold>{`We're live.`} </UIText>
-                            Watch now!
-                          </UIText>
-                        }
-                      />
-                    </CardContent>
-                  </LiveCard>
-                </TouchableScale>
-              )}
-            </Mutation>
-          )}
+          {({ track }) => {
+            return liveStream
```

Comment:
```suggestion
            return liveStream.media.sources.length
```
or something like that. default check all the way through or use lodash.